gaia-list.js

```diff
@@ -82,15 +85,20 @@ module.exports = component.register('gaia-list', {
     this.els.inner.appendChild(els.container);
 
     // var reflow = els.ripple.offsetTop;
-    var scale = pos.item.width / 1.2;
+    var scale = itemWidth / 1.2;
+    // tap at edge of the list
+    if ((itemWidth - scale) > (clientX - itemLeft) ||
+      (itemWidth - scale) > (pos.item.right - clientX)) {
+      scale = itemWidth;
+    }
     var duration = 500;
 
-    setTimeout(function() {
+    requestAnimationFrame(function() {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame

In summary, use `requestAnimationFrame` instead of `setTimeout` can boost animation performance.